### ISSUE TITLE
x/tools/gopls: consider adding special cases for main Go tree and golang.org/x modules when linkifying issues

### ISSUE THREAD

**Issue by dmitshur** [golang/go#46245]
@heschi [pointed out](https://go-review.googlesource.com/c/build/+/321011/1/cmd/release/release.go#279) to me that `gopls` has a feature of turning text like `org/repo#123` into links to corresponding GitHub issues.

The main Go repository and most (if not all) golang.org/x modules use the main Go issue tracker (https://golang.org/issues) and have many¹ existing comments that refer to Go issues as just "#​123" or "issue 123".

If it's deemed okay to add special cases to that behavior for the Go project itself, this is a feature to adjust the linkifying behavior of `gopls` to recognize when the module path is something in the standard library (just `std`, `cmd` would go a long way) or has a `golang.org/` prefix, and not require the "golang/go" prefix. That would help make existing comments useful and reduce the desire to be adding the otherwise-not-strictly-needed "golang/go" prefix in Go comments.

<details><summary>¹ (Details)</summary><br>

A quick grep in the Go tree at commit a2c07a9a1a445c28fa4db9bd1472b28cad3ca818:

```
"see issue \d+"  - 286 matches
"see issue #\d+" - 143 matches
"see #\d+"       - 82 matches
"golang/go#\d+"  - 5 matches
```

</details>

CC @stamblerre, @heschi.

**Comment by findleyr** [comment]
If we're going to do this, couldn't we equally well add support for any module path that is a subpath of `github.com/<user>/<repo>` (i.e. resolve `#123` to `github.com/<user>/<repo>/issues/123`)?

Of course we'd still need special handling for golang.org modules.

**Comment by dmitshur** [comment]
Yes, I think that would be possible too, although there may be some edge cases that couldn't be handled easily. For example, GitHub forks, or repositories with custom module paths (`module example.com/m`) whose source code lives on github.com.

If considering GitHub, might want to also consider other common code hosting sites that `cmd/go` already [knows about](https://github.com/golang/go/blob/15a374d5c1336e9cc2f8b615477d5917e9477440/src/cmd/go/internal/vcs/vcs.go#L1178) and their issue trackers. Of course such work would be larger and should be tracked in separate issue(s), so this one can be remain about just the Go project.

**Comment by heschi** [comment]
We could consider keying off of `codereview.cfg` as a (theoretically) less special case than golang.org import paths, but AFAIK literally nobody else uses `git-codereview`.
